tests/operator/helper.go

```diff
@@ -45,3 +50,34 @@ func SplitCsv(csv string) CsvResult {
 	}
 	return result
 }
+
+func isInstallModeSingleNamespace(installModes []v1alpha1.InstallMode) bool {
+	for i := 0; i < len(installModes); i++ {
+		if installModes[i].Type == v1alpha1.InstallModeTypeSingleNamespace {
+			return true
+		}
+	}
+	return false
+}
+
+func findOperatorGroup(name, namespace string, groups []*v1.OperatorGroup) *v1.OperatorGroup {
+	for _, group := range groups {
+		if group.Name == name && group.Namespace == namespace {
+			return group
+		}
+	}
+	return nil
+}
+
+func checkOperatorInstallationCompliance(operatorGroup *v1.OperatorGroup, operatorNamespace string, targetNamespaces []string, isSingleNamespaceInstallMode bool) bool {
+	opGroupTargetNamespaces := operatorGroup.Spec.TargetNamespaces
+	if isSingleNamespaceInstallMode {
+		return len(opGroupTargetNamespaces) == 1 && len(targetNamespaces) == 1 && opGroupTargetNamespaces[0] == targetNamespaces[0]
+	}
+	for _, ns := range opGroupTargetNamespaces {
+		if ns == operatorNamespace {
+			return false
+		}
+	}
+	return true
```

Comment:
Is `opGroupTargetNamespaces` a []string?

Then we could use `stringhelper.StringInSlice` and remove this loop.

Reply:
Yes, []string. 
I don't think we need this function. The definition uses Contains function, which is like substring check, whereas we need here Equals check on strings. StringInSlice would be true for namespace abc if the slice contains xyzabcd.

Reply:
Just set the `contains` parameter to false for StringInSlice.